www/src/pages/editor/SearchBar.tsx

```diff
@@ -185,59 +198,115 @@ export const SearchBar: React.FC<SearchBarProps> = observer(({ searchArgs, setSe
-        Add Mineral Site
-      </Button>
-    </Flex>
+    <div>
+      <Flex gap="small" justify="space-between" align="center">
+        <Space>
+          <Typography.Text className={styles.label}>
+            Commodity<span style={{ color: "red" }}>*</span>:
+          </Typography.Text>
+          <Select
+            style={{ width: 200 }}
+            value={searchArgs.commodity}
+            placeholder="Select a commodity"
+            showSearch={true}
+            optionFilterProp="label"
+            onChange={(id: string) =>
+              setSearchArgs({
+                ...searchArgs,
+                commodity: commodityStore.get(id)!.name,
+              })
+            }
+            options={commodityOptions}
+          />
+          <Typography.Text className={styles.label}>Deposit Type:</Typography.Text>
+          <Select
+            style={{ width: 200 }}
+            allowClear={true}
+            value={searchArgs.depositType}
+            placeholder="Select a deposit type"
+            showSearch={true}
+            optionFilterProp="label"
+            onChange={(id?: string) =>
+              setSearchArgs({
+                ...searchArgs,
+                depositType: id === undefined ? undefined : depositTypeStore.get(id)!.name,
+              })
+            }
+            options={depositTypeOptions}
+          />
+          <Typography.Text className={styles.label}>Country:</Typography.Text>
+          <Select
+            style={{ width: 150 }}
+            allowClear={true}
+            value={searchArgs.country}
+            placeholder="Select a country"
+            showSearch={true}
+            optionFilterProp="label"
+            onChange={(id?: string) =>
+              setSearchArgs({
+                ...searchArgs,
+                country: id === undefined ? undefined : countryStore.get(id)!.name,
+              })
+            }
+            options={countryOptions}
+          />
+          <Typography.Text className={styles.label}>State/Province:</Typography.Text>
+          <Select
+            style={{ width: 150 }}
+            allowClear={true}
+            value={searchArgs.stateOrProvince}
+            placeholder="Select a state/province"
+            showSearch={true}
+            optionFilterProp="label"
+            onChange={(id?: string) =>
+              setSearchArgs({
+                ...searchArgs,
+                stateOrProvince: id === undefined ? undefined : stateOrProvinceStore.get(id)!.name,
+              })
+            }
+            options={stateOrProvinceOptions}
+          />
+        </Space>
+        <Space>
+          <Button type="primary" onClick={onOpenNewMineralSiteForm}>
+            Add Mineral Site
+          </Button>
+          <Button type="primary" icon={<SettingOutlined />} onClick={() => setVisible(true)}></Button>
+
+          <Modal title="Add new field" open={visible} onCancel={() => setVisible(false)} footer={null} width="70%">
```

Comment:
Create a separate component for this modal, and handle setVisible there. We should move setVisible to the settingStore to avoid communicating between Button & Modal